tests/Bridge/Doctrine/Orm/Filter/BooleanFilterTest.php

```diff
@@ -0,0 +1,250 @@
+    }
+
+    public function testGetDescriptionDefaultFields()
+    {
+        $filter = new BooleanFilter($this->managerRegistry, new RequestStack());
+        $this->assertEquals([
+            'dummyBoolean' => [
+                'property' => 'dummyBoolean',
+                'type' => 'boolean',
+                'required' => false,
+            ],
+        ], $filter->getDescription($this->resourceClass));
+    }
+
+    /**
+     * Providers 3 parameters:
+     *  - filter parameters.
+     *  - properties to test. Keys are the property name. If the value is true, the filter should work on the property,
+     *    otherwise not.
+     *  - expected DQL query.
+     *
+     * @return array
+     */
+    public function filterProvider()
+    {
+        return [
+            // test with true value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => 'true',
+
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with false value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => 'false',
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with non-boolean value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => 'toto',
+                ],
+                sprintf('SELECT o FROM %s o', Dummy::class),
+            ],
+            // test with 0 value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => '0',
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with 1 value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => '1',
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with nested properties.
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'relatedDummy.id' => null],
+                ],
+                [
+                    'relatedDummy.id' => '1',
```

Comment:
This does not show that it works with nested properties, because the field type is wrong so it's just being skipped.